src/server.rs

```diff
@@ -1073,6 +1088,10 @@ impl Server {
                 'H' => {
                     self.in_copy_mode = true;
                     self.data_available = true;
```

Comment:
Is this line still meant to be here?
If so, the following conditional assignment is unnecessary.